kong/plugins/oidc/handler.lua

```diff
@@ -29,27 +29,52 @@ function CustomHandler:access(config)
 
     session.configure(config)
 
-    local res, err = require("resty.openidc").authenticate(oidcConfig)
+    if tryIntrospect(oidcConfig) then
```

Comment:
Those nested if/else statements smell a bit. I would create a few functions, so it 

> [...] reads like well-written prose

Reply:
Please, reveal us those few functions you would write instead of nested ifs.

Reply:
created doAuthentication function to make code more clear